backend/infrahub/graphql/mutations/account.py

```diff
@@ -106,33 +107,33 @@ async def create_token(
     @classmethod
     @retry_db_transaction(name="account_token_delete")
     async def delete_token(
-        cls, db: InfrahubDatabase, account: Node, data: Dict[str, Any], info: GraphQLResolveInfo
+        cls, db: InfrahubDatabase, account: CoreNode, data: Dict[str, Any], info: GraphQLResolveInfo
     ) -> Self:
         token_id = str(data.get("id"))
 
         results = await NodeManager.query(
-            schema=InfrahubKind.ACCOUNTTOKEN, filters={"account_ids": [account.id], "ids": [token_id]}, db=db
+            schema=InternalAccountToken, filters={"account_ids": [account.id], "ids": [token_id]}, db=db
         )
 
         if not results:
             raise NodeNotFoundError(node_type="AccountToken", identifier=token_id)
 
         async with db.start_transaction() as dbt:
-            await results[0].delete(db=dbt)
+            await results[0].delete(db=dbt)  # type: ignore[arg-type]
 
         return cls(ok=True)  # type: ignore[call-arg]
 
     @classmethod
     @retry_db_transaction(name="account_update_self")
     async def update_self(
-        cls, db: InfrahubDatabase, account: Node, data: Dict[str, Any], info: GraphQLResolveInfo
+        cls, db: InfrahubDatabase, account: CoreNode, data: Dict[str, Any], info: GraphQLResolveInfo
     ) -> Self:
         for field in ("password", "description"):
             if value := data.get(field):
                 getattr(account, field).value = value
 
         async with db.start_transaction() as dbt:
-            await account.save(db=dbt)
+            await account.save(db=dbt)  # type: ignore[arg-type]
```

Comment:
Same comment as with .delete(). Shouldn't this work without the ignore?

Reply:
ignore is required here because the Protocol for `InfrahubDatabase` doesn't match the real implementation right now so mypy is complaining that we are passing an object that isn't compatible.
The real challenge is actually with `SchemaBranch` which just has a placeholder Protocol for now.
I looked into improving the protocol for `SchemaBranch` but it has a LOT of methods so I decided to skip that for now.
Would be awesome if there was a tool to automatically build Protocol for a given class :)